core/client/MeetInTheMiddle.md

```diff
@@ -0,0 +1,43 @@
+# Meet In The Middle Algorithm
```

Comment:
Consider moving this to the design docs folder, and adding a link to it from the code comments.

Reply:
done